src/context/authContext/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase/firebase";
+import { useContext, useEffect, useState } from "react";
+
+const AuthContext = React.createContext();
+
+export function useAuth() {
+  return useContext(AuthContext);
+}
+
+export function AuthProvider({ children }) {
+  const [currentUser, setCurrentUser] = useState(null);
+  const [userLoggedIn, setUserLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, initializeUser);
+    return unsubscribe;
+  }, []);
+
+  async function initializeUser(user) {
+    if (user) {
+      setCurrentUser({ ...user });
+      setUserLoggedIn(true);
+    } else {
+      setCurrentUser(null);
+      setUserLoggedIn(false);
+    }
+    setLoading(false);
+  }
+  const value = {
+    currentUser,
+    userLoggedIn,
+    loading,
+  };
+
+  return (
+    <AuthContext.Provider value={value}>
+      {!loading && children}
+    </AuthContext.Provider>
+  );
```

Comment:
Instead of blocking whole app from rendering, can we navigate the user to the relevant page when they're logged in? They can get navigated to auth page if they're not logged in for example. Therefore "loading" state can be removed here.

Reply:
To resolve this, useNavigate has been employed.